packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts

```diff
@@ -70,8 +68,10 @@ export const logOpenGeneratedAnswerSource = (
     analyticsType: 'Rga.CitationClick',
     analyticsPayloadBuilder: (state): Rga.CitationClick | undefined => {
       const citation = citationSourceSelector(state, citationId);
+      const {generativeQuestionAnsweringId} =
+        generativeQuestionAnsweringIdSelector(state);
       return {
-        answerId: state.search?.response.searchUid || '',
+        answerId: generativeQuestionAnsweringId ?? '',
```

Comment:
It's a bit dumb when you think about it too, because an empty string is an invalid event.

But I guess it's okay for now.